daml-lf/spec/contract-id.rst

```diff
@@ -0,0 +1,245 @@
+.. Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+V1 Contract ID allocation scheme
+================================
+
+Goals
+^^^^^
+
+* Allows ordering contract IDs and make DAML semantics depend on this
+  order, e.g., for comparison builtin and maps using IDs as keys.
+* Eliminate all contract ID translations for central commiter ledger
+
+Requirements
+^^^^^^^^^^^^
+
+**Executability**: DAML engine can determine ordering of the contract
+IDs before it hands out the transaction to the write service.
+
+**Validation**: The allocation scheme commutes with transaction
+projection. That is, the contract IDs for contracts created in the
+projection of a transaction to a set of parties can be computed
+solely from the projection and input seeds.
```

Comment:
This is the first time in this document that you mention "input seeds". Can you explain them briefly here or put a forward reference to their definition?

Reply:
I drop the "input seeds". I just consider the seed of the node to be part of the projection.